### PR TITLE
[Nereids] add leading tpc-h

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

add tpc-h shape cases using leading hint
except:
- single table without join q1 q6
- not support feature include tables after subquery unnested q2 q16 q18 q20 q21 q22

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by englefly — DISMISSED** [review]

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

File: regression-test/suites/nereids_hint_tpch_p0/shape/q10.groovy [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+
+
+    
+sql 'set be_number_for_test=3'
+
+    qt_select """
+    explain shape plan
+    select 
+    /*+ leading(lineitem {{customer orders} nation}) */
```
Comment (xzj7019) [inline regression-test/suites/nereids_hint_tpch_p0/shape/q10.groovy:38]: only control the leading, can we add shuffle control also to verify them both?

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

File: regression-test/data/nereids_hint_tpch_p0/shape/q10.out [hunk 2]
```diff
@@ -2,33 +2,34 @@
...
-----------------------------PhysicalOlapScan[nation]
+--------------------------PhysicalDistribute[DistributionSpecReplicated]
+----------------------------PhysicalProject
+------------------------------PhysicalOlapScan[nation]
 
-Used: leading(lineitem { { customer orders } nation })
-UnUsed:
+Hint log:
+Used:  leading(lineitem { { customer shuffle orders } broadcast nation } )
+UnUsed: [shuffle]_2
```
Comment (xzj7019) [inline regression-test/data/nereids_hint_tpch_p0/shape/q10.out:33]: why this "UnUsed" exists?

File: regression-test/data/nereids_hint_tpch_p0/shape/q16.out [hunk 3]
```diff
@@ -2,25 +2,26 @@
...
-----------------------PhysicalOlapScan[supplier]
+--------------------filter(( not (p_brand = 'Brand#45')) and ( not (p_type like 'MEDIUM POLISHED%')) and p_size IN (14, 19, 23, 3, 36, 45, 49, 9))
+----------------------PhysicalOlapScan[part]
 
-Used: leading(partsupp part supplier)
-UnUsed:
-SyntaxError:
+Hint log:
+Used:
+UnUsed: [shuffle]_2
```
Comment (xzj7019) [inline regression-test/data/nereids_hint_tpch_p0/shape/q16.out:25]: this UnUsed?

File: regression-test/data/nereids_hint_tpch_p0/shape/q5.out [hunk 4]
```diff
@@ -2,24 +2,43 @@
...
+------------------------------------------PhysicalOlapScan[region]
+------------------PhysicalDistribute[DistributionSpecHash]
+--------------------PhysicalProject
+----------------------PhysicalOlapScan[customer]
 
-Used: leading(lineitem { supplier { nation region } } orders customer)
-UnUsed:
+Hint log:
+Used:  leading(lineitem orders { supplier broadcast { nation broadcast region } } shuffle customer )
+UnUsed: [broadcast]_2
```
Comment (xzj7019) [inline regression-test/data/nereids_hint_tpch_p0/shape/q5.out:42]: this one

File: regression-test/data/nereids_hint_tpch_p0/shape/q7.out [hunk 5]
```diff
@@ -1,44 +1,44 @@
...
-------------------------------------PhysicalOlapScan[lineitem]
+--------------------------------------filter(n_name IN ('FRANCE', 'GERMANY'))
+----------------------------------------PhysicalOlapScan[nation]
 
-Used:
-UnUsed:
-SyntaxError: leading(customer { orders { lineitem { supplier { n1 n2 } } } }) Msg:tables should be same as join tables
+Hint log:
+Used: leading(lineitem { supplier broadcast n1 } { orders { customer broadcast n2 } } )
+UnUsed: [broadcast]_2
```
Comment (xzj7019) [inline regression-test/data/nereids_hint_tpch_p0/shape/q7.out:42]: this one

File: regression-test/data/nereids_hint_tpch_p0/shape/q9.out [hunk 6]
```diff
@@ -2,44 +2,45 @@
...
+----------------------------------PhysicalOlapScan[nation]
+------------------PhysicalDistribute[DistributionSpecHash]
 --------------------PhysicalProject
 ----------------------PhysicalOlapScan[partsupp]
 
-Used: leading(orders { { lineitem part } { supplier nation } } partsupp)
-UnUsed:
+Hint log:
+Used:  leading(orders { lineitem shuffle part } shuffle { supplier broadcast nation } shuffle partsupp )
+UnUsed: [shuffle]_2
```
Comment (xzj7019) [inline regression-test/data/nereids_hint_tpch_p0/shape/q9.out:44]: this one

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buidall

**Review by xzj7019 — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Comment by morrySnow** [comment]
run buildall

**Pull request merged**